src/app/components/file-upload/file-upload.component.ts

```diff
@@ -1,40 +1,156 @@
 /* eslint-disable no-console */
 /* eslint-disable no-undef */
-import { Component, OnInit } from '@angular/core';
+import {
+  Component, OnInit, ViewChild, ElementRef
+} from '@angular/core';
 import { FormBuilder, FormGroup } from '@angular/forms';
 import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import yaml from 'js-yaml';
+
+// NgRX
+import { Store } from '@ngrx/store';
+import { Router } from '@angular/router';
+import { SwagPaths } from '../../state/swag-paths/swag-paths.model';
+import { POPULATE_PATHS } from '../../state/swag-paths/swag-paths.actions';
 
 @Component({
   selector: 'app-file-upload',
   templateUrl: './file-upload.component.html',
   styleUrls: ['./file-upload.component.scss'],
 })
 export class FileUploadComponent implements OnInit {
-  private baseUrl = 'http://localhost:8083/Docutest';
+  // NgRX
+  swagPaths$: Observable<SwagPaths>;
+
+  private Base_Url = 'http://localhost:8083/Docutest';
 
   public uploadForm: FormGroup;
 
-  constructor(private formBuilder: FormBuilder, private http: HttpClient) {}
+  public selectedFile;
+
+  public errorMsg: String = '';
+
+  public show = false;
+
+  /* Used to reset the file input */
+  @ViewChild('fileIn')
+  fileInput: ElementRef;
+
+  swag;
+
+  constructor(private formBuilder: FormBuilder, private http: HttpClient,
+    private store: Store<{swagPaths}>, private router: Router) {
+    this.swagPaths$ = store.select('swagPaths');
+  }
 
   ngOnInit(): void {
     this.uploadForm = this.formBuilder.group({
       swaggerFile: [''],
     });
   }
 
+  /*
```

Comment:
These useful comments here could actually use `/** comment */` syntax, which are called JSDoc comments. Slightly more helpful for intellisense.

Reply:
This has yet to be resolved.